### PR TITLE
Feat/refactor core connect

### PR DESCRIPTION
**Pull request by zachferland**
~~draft, literally have not run it, except tests~~

### CODE REVIEW

File: connect/src/__tests__/manage3IDs.test.ts [hunk 1]
```diff
@@ -55,6 +58,22 @@ describe('3ID Management', () => {
 
   test('creates/loads did from in memory', async () => {})
 
+  //TODO This test fails, if you log the authclient and authserver you can see it returns
```
Comment (zachferland) [inline connect/src/__tests__/manage3IDs.test.ts:61]: @PaulLeCam mainly the test I wanted to try, and couldnt get to work. Also start using elsewhere but more or less just wrote everything and didnt actually step through to make sure things are working yet, getting there tmrrw still. just sharing for context, no need to prio or anything

File: connect/src/authProviderRelay.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  client: RPCClient
+  readonly isAuthProvider = true;
+
+  constructor(target:PostMessageTarget) {
+    target = target || window.parent
+    const transport = new PostMessageTransport<AuthProviderMethods>(window, target, {
+      postMessageArguments: [window.origin], 
+    })
+    //TODO Cant resolve these types???
+    this.client = createClient<AuthProviderMethods>(transport)
```
Comment (zachferland) [inline connect/src/authProviderRelay.ts:32]: @PaulLeCam  any idea why i cant resolve types properly here? ( transport)

Comment (PaulLeCam) [reply connect/src/authProviderRelay.ts:32]: Yes the transport only cares about messages in/out so the methods are too high-level here, it needs something like:
```ts
type Request = RPCRequest<AuthProviderMethods, keyof AuthProviderMethods>
type Response = RPCResponse<AuthProviderMethods, keyof AuthProviderMethods>
const transport = new PostMessageTransport<Response, Request>(window, target, {
  postMessageArguments: [window.origin],
})
```

I'm working on updating this in a separate branch off this one, as we also need to use the latest version of rpc-utils to have the proper types, hopefully that should unblock you.

**Comment by zachferland** [comment]
@PaulLeCam  all set here for now, management lib and authprovider rpc included, with test passing now 

still going to remove/clean up one file or two and add some more unit tests, but will include in next pr

**Review by PaulLeCam — APPROVED** [review]
Great, thanks!

**Pull request merged**